plugins/woocommerce/includes/abstracts/abstract-wc-payment-gateway.php

```diff
@@ -562,3 +562,79 @@ public function add_payment_method() {
 		);
 	}
 }
+
+/**
+ * Email the site admin when a payment gateway is enabled.
+ *
+ * @param string $option Option name.
+ * @param array  $old_value Old value.
+ * @param array  $value New value.
+ * @since 8.4.0
+ */
+function wc_notify_payment_gateway_enabled( $option, $old_value, $value ) {
+	$gateway_option_keys = array();
+	$wc_payment_gateways = WC_Payment_Gateways::instance()->payment_gateways();
+	foreach ( $wc_payment_gateways as $gateway ) {
+		$gateway_option_keys[] = $gateway->get_option_key();
+	}
+	if ( ! in_array( $option, $gateway_option_keys, true ) ) {
+		return;
+	}
```

Comment:
This might be seen as a micro-optimization and it definitely is not essential, but an alternative could be:

```suggestion
	foreach ( WC_Payment_Gateways::instance()->payment_gateways() as $gateway ) {
		if ( $option === $gateway->get_option_key() ) {
			$match = true;
			break;
		}
	}

	if ( ! isset( $match ) ) {
		return;
	}
```

(Imho, just as readable: but potentially fewer iterations and var assignments.)

Reply:
👍 addressed in 6417bdf.